_posts/2017-03-05-artsy-technology-stack-2017.md

```diff
@@ -0,0 +1,165 @@
+
+We've also explored alternate communication patterns, so systems aren't as dependent on each other's APIs. Recently we've begun publishing a stream of interesting data events from our core systems. Other systems can simply subscribe to the notifications they care about, so the source system doesn't need to be concerned about integrating with one more destination. After experimenting with [Kafka](https://kafka.apache.org) but finding it hard to manage, we switched to [RabbitMQ](https://www.rabbitmq.com) for this purpose. To provide consistency when publishing events we have [our own gem](https://github.com/artsy/artsy-eventservice).
+
+## Operations
+
+All our recent AWS infrastructure is configured in code using [Terraform](https://www.terraform.io). This approach has allowed us to quickly replicate entire deployments along with their dependencies and has increased visibility into the state of our infrastructure across our teams. We started developing an open source [Docker](https://www.docker.com) workflow toolkit named [Hokusai](https://github.com/artsy/hokusai) in order to manage a containerized workflow, CI and deployment to [Kubernetes](https://kubernetes.io). Our Kubernetes clusters are managed using [Kops](https://github.com/kubernetes/kops) and similarly provisioned using Terraform. This new workflow is reducing our dependence on Heroku, giving us more flexibility in our deployments and a more efficient use of server resources.
+
+# People and Culture
+
+## Open Source by Default
+
+By the end of 2016, almost every major front-end application at Artsy was [Open Source by Default](http://code.dblock.org/2015/02/09/becoming-open-source-by-default.html). This means looking for reasons for something to be closed-source as opposed to reasons to be open. Our entire working process is done in the open, from developer PRs to QA. This post was also written collaboratively and in the open as you can see [here](https://github.com/artsy/artsy.github.io/pull/325).
+
+Sometimes it makes sense to keep some details private for competitive reasons. We therefore also create a private GitHub repository for front-end teams that require cross-project issues and team milestones. This is done using [ZenHub](https://www.zenhub.com), and is managed by Engineering leads and Product Managers.
+
+## Developer Workflow
+
+Most development workflow tries to mimic large open-source project development where most work happens on forks and is pull-requested into an Artsy repository shared by everyone. Typically an engineer will start a project, application or service and is automatically its benevolent dictator. They will add continuous integration and will ask other engineers on the team to code review everything from day one. Others will join and entire teams may take over. Continuous deployment and other operational infrastructure will get setup early.
```

Comment:
> where most work happens on forks

That’s really dependent on the team. Don’t know if it needs to be included at all, but if there is reason for it then it might need amending.